core/services/relay/evm/mercury/data_source.go

```diff
@@ -63,7 +70,35 @@ func (ds *datasource) Observe(ctx context.Context, repts ocrtypes.ReportTimestam
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			obs.MaxFinalizedBlockNumber.Val, obs.MaxFinalizedBlockNumber.Err = ds.fetcher.FetchInitialMaxFinalizedBlockNumber(ctx)
+			val, err := ds.fetcher.FetchInitialMaxFinalizedBlockNumber(ctx)
+			if err != nil {
+				obs.MaxFinalizedBlockNumber.Err = err
+				return
+			}
+			if val != nil {
+				obs.MaxFinalizedBlockNumber.Val = *val
+				return
+			}
+			if ds.initialBlockNumber == nil {
+				if obs.CurrentBlockNum.Err != nil {
+					obs.MaxFinalizedBlockNumber.Err = fmt.Errorf("FetchInitialMaxFinalizedBlockNumber returned empty LatestReport; this is a new feed. No initialBlockNumber was set, tried to use current block number to determine maxFinalizedBlockNumber but got error: %w", obs.CurrentBlockNum.Err)
+				} else {
+					// Subract 1 here because we will later add 1 to the
+					// maxFinalizedBlockNumber to get the first validFromBlockNum, which
+					// ought to be the same as current block num.
+					obs.MaxFinalizedBlockNumber.Val = obs.CurrentBlockNum.Val - 1
+					ds.lggr.Infof("FetchInitialMaxFinalizedBlockNumber returned empty LatestReport; this is a new feed so maxFinalizedBlockNumber=%d (initialBlockNumber unset, using currentBlockNum=%d-1)", obs.MaxFinalizedBlockNumber.Val, obs.CurrentBlockNum.Val)
+				}
+			} else {
+				// NOTE: It's important to return -1 if the server is missing any past
```

Comment:
```suggestion
				// NOTE: It's important to subtract 1 if the server is missing any past
```
Otherwise it reads like you must return -1